doc/source/reference/c-api.iterator.rst

```diff
@@ -761,17 +765,19 @@ Construction and Destruction
 
 .. c:function:: int NpyIter_Close(NpyIter* iter)
 
-    Resolves any needed writeback resolution. Must be called before
-    ``NpyIter_Deallocate``. After this call it is not safe to use the operands.
+    Resolves any needed writeback resolution. Should be called before
+    :c:func::`NpyIter_Deallocate`. After this call it is not safe to use the operands.
+    When using :c:func:`NpyIter_Copy`, only one call to :c:func:`NpyIter_Close`
```

Comment:
Is `NpyIter_Copy` exposed to python?

Reply:
yes, as [`it.copy()`](http://www.numpy.org/devdocs/reference/generated/numpy.nditer.copy.html#numpy.nditer.copy).  Should I expand the documentation of `it.copy` to mention writeback considerations?

Reply:
Won'y `np.nditer.__del__` complain if you only call `close` on the original and not the copy?

Reply:
no, the complaint is on the operands, which are shared:
```
def test_copy():
    a = np.arange(6, dtype='f4')
    au = a.byteswap().newbyteorder()
    it1 = np.nditer(au, [], [['readwrite', 'updateifcopy']],
                        casting='equiv', op_dtypes=[np.dtype('f4')])
    it2 = it1.copy()
    x = it1.operands[0]
    assert x is it2.operands[0]
    assert x.flags.writebackifcopy is True
    it1.close()
    assert x.flags.writebackifcopy is False
```

commenting the last two lines *does* emit a warning

Reply:
Not sure what you mean. The context manager resolves operands on exit:
```
def test_copy():
    a = np.arange(6, dtype='f4')
    au = a.byteswap().newbyteorder()
    it1 = np.nditer(au, [], [['readwrite', 'updateifcopy']],
                        casting='equiv', op_dtypes=[np.dtype('f4')])
    with it1:
        it2 = it1.copy()
        x = it1.operands[0]
        assert x is it2.operands[0]
        assert x.flags.writebackifcopy is True
    assert x.flags.writebackifcopy is False
    # no warning on it2 deallocate, the operand is resolved via it1
```
Note that nesting the context managers, `it1` is still open even though `it2` is closed
```
def test_copy():
    a = np.arange(6, dtype='f4')
    au = a.byteswap().newbyteorder()
    it1 = np.nditer(au, [], [['readwrite', 'updateifcopy']],
                        casting='equiv', op_dtypes=[np.dtype('f4')])
    with it1:
        with it1.copy() as it2:
            x = it1.operands[0]
            assert x is it2.operands[0]
            assert x.flags.writebackifcopy is True
        assert x.flags.writebackifcopy is False
        y = it1.operands[0] # can still access it1's operand, it is not closed
        assert y is x
```

Reply:
Why do we allow access to `it1.operands` and forbid access to `it2.operands` on that line, when they point to the same object? Either they're both invalid, or neither of them are invalid.

Reply:
Yep, this PR looks good to me as a doc improvement, but I'm worried that writebackifcopy might have been the wrong design after all. In my opinion, you should have to call close on _both_ iterators for writebacks to be resolved, which would require a reference count.

Reply:
note added to issue #10958